### PR TITLE
GeeksForGeeks Tests

### PR DESCRIPTION
**Pull request by AaronLopes**
(Ignore few SO tests for now, too many API calls today)

### CODE REVIEW

**Comment by anbenson** [comment]
Remind me who's already approved this? Is it just me who's left, or should we be waiting for the pod to approve and so forth?

**Comment by irisliu77** [comment]
> Remind me who's already approved this? Is it just me who's left, or should we be waiting for the pod to approve and so forth?

the pod approved it, i think Joan also needs to be added here

File: src/test/java/com/google/step/snippet/external/StackOverflowClientTest.java [hunk 1]
```diff
@@ -100,7 +100,7 @@ public void noAnswerSearchTest() {
     Card actual =
         client.search(
             "https://stackoverflow.com/questions/44686609/implementing-a-neural-network-in-haskell",
-            "implement neural net in haskell");
+            "implement neural network in haskell");
```
Comment (anbenson) [inline src/test/java/com/google/step/snippet/external/StackOverflowClientTest.java:103]: Was this an intended change?

File: src/test/java/com/google/step/snippet/external/GeeksForGeeksClientTest.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public final class GeeksForGeeksClientTest {
+
+  private final GeeksForGeeksClient client = new GeeksForGeeksClient("CSE_ID");
+
+  @Test
+  public void htmlCodeCard() {
```
Comment (joanchenx) [inline src/test/java/com/google/step/snippet/external/GeeksForGeeksClientTest.java:17]: I think the logic for determining when to do HTML escaping should also be checked. Could you add tests where you check:
1. Query does not contain escape filter, but webpage does. In this case, the HTML should still be escaped. 
Example: `client.search("https://www.geeksforgeeks.org/how-to-change-the-height-of-br-tag/", "br tag");`
2. Query contains escape filter, but webpage does not.  In this case, the HTML should still be escaped. 
Example: `client.search("https://www.geeksforgeeks.org/how-to-display-error-without-alert-box-using-javascript/", "html display error");`
3. Neither query nor webpage contains escape filter. In this case, the HTML should not be escaped. 
Example: `client.search("https://www.geeksforgeeks.org/how-to-display-error-without-alert-box-using-javascript/", "javascript display error");`

Comment (AaronLopes) [reply src/test/java/com/google/step/snippet/external/GeeksForGeeksClientTest.java:17]: Sure!

File: src/test/java/com/google/step/snippet/external/StackOverflowClientTest.java (hunk 1 above)
Comment (AaronLopes) [reply src/test/java/com/google/step/snippet/external/StackOverflowClientTest.java]: No, I think it was from a previous edit I made to the file. Removed.

File: src/test/java/com/google/step/snippet/external/GeeksForGeeksClientTest.java [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+                + " </div>  Submit  \n"
+                + "</div>",
+            "https://www.geeksforgeeks.org/how-to-display-error-without-alert-box-using-javascript/",
+            "Errors in JavaScript can be displayed without the use of alert boxes but using the"
+                + " alert box is the traditional way to do that. We can show errors with two"
+                + " methods without using the alert box.",
+            "GeeksForGeeks",
+            "https://geeksforgeeks.org/favicon.ico");
+    assertEquals(expected, actual);
+  }
```
Comment (joanchenx) [inline src/test/java/com/google/step/snippet/external/GeeksForGeeksClientTest.java:182]: Great! Small nit: could you move these underneath the `htmlCodeCard` test?

**Pull request merged**